core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -1045,29 +1037,31 @@ private static void shiftMapping(Map<Integer, Integer> mapping, int startIndex,
   }
 
   protected @Nullable Frame decorrelateFetchOneSort(Sort sort, final Frame frame) {
-    Frame aggFrame = decorrelateSortAsAggregate(sort, frame);
-    if (aggFrame != null) {
-      return aggFrame;
+    if (sort.offset == null
+        && sort.fetch != null
+        && RexLiteral.intValue(sort.fetch) == 0) {
```

Comment:
is 0 even allowed?
Also, intValue seems unsafe; there have been some bug fixes about this in the past.
BigInt is probably safest.

Reply:
The purpose of this if check is to skip the subsequent optimization process.